File: server/router/emag/emag_product.go
package emag

import (
	"github.com/gin-gonic/gin"
)

type EmagProductRouter struct{}

// InitEmagProductRouter 初始化产品路由
func (e *EmagProductRouter) InitEmagProductRouter(Router *gin.RouterGroup) {
	emagProductRouter := Router.Group("emagProduct")
	{
		emagProductRouter.GET("getList", emagProductApi.GetProductList)        // 分页获取产品列表
		emagProductRouter.POST("sync", emagProductApi.SyncProducts)            // 同步产品
		emagProductRouter.GET("getStatusList", emagProductApi.GetStatusList)   // 获取状态列表
		emagProductRouter.GET("getCountryList", emagProductApi.GetCountryList) // 获取国家列表
	}
}

File: web/src/api/emagProduct.js
import service from '@/utils/request'

// 分页获取产品列表
export const getProductList = (params) => {
  return service({
    url: '/emagProduct/getList',
    method: 'get',
    params
  })
}

// 同步产品
export const syncProducts = (data) => {
  return service({
    url: '/emagProduct/sync',
    method: 'post',
    data
  })
}

// 获取状态列表
export const getStatusList = () => {
  return service({
    url: '/emagProduct/getStatusList',
    method: 'get'
  })
}

// 获取国家列表
export const getCountryList = () => {
  return service({
    url: '/emagProduct/getCountryList',
    method: 'get'
  })
}
